src/price_data.py

```diff
@@ -119,8 +119,22 @@ def _get_price_binance(
         response.raise_for_status()
 
         if len(data) == 0:
-            log.warning("Binance offers no price for `%s` at %s", symbol, utc_time)
-            return decimal.Decimal()
+            log.warning(
+                f"""
+            Binance offers no price
+            for {symbol} at {utc_time}
+            Retrying with {base_asset}USDT and {quote_asset}USDT
+            """,
+                symbol,
+                utc_time,
```

Comment:
Use %-formatting in logging. Check other logging messages , too.

Reply:
Its mixed actually. Line 189 for example. Can you explain the reasoning, so I can infer the correct formatting?

Reply:
%-formatting is lazy evaluated and postpones the string calculation of the argument after the check whether the logging message is necessary (e.g. logging level is warning, a log to debug isn't evaluated).

Most string calculations are very cheap, so it's not that of an performance issue, but I am still hanging on to it, so I never could have the performance problem.